### PR TITLE
Add Donut Graphs for Repository Line Makeup 

### PR DESCRIPTION
**Pull request by IsaacMilarky**
## Problem

There was no line composition graph to match the line composition metrics that were recently added. 

There was also a minor bug in the issue gathering logic. 

## Solution

Add another graph generation method to create a PyGals Pie chart with an inner radius. The values of the line composition are calculated from the [previously gathered metrics](#95). 

## Result

Alter the class definitions for `RangeMetric` and `ListMetric` to fix a niche bug where tuples were combined in the larger list generated as is the behavior for `RangeMetric`.

Add the aforementioned Pie graph with the proper radius set. The three pie sections are: comments, blank lines, and other. Here is an example of the donut graph:
total_line_makeup_metrics_data

### CODE REVIEW

File: scripts/gen_graphs.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ def generate_all_graphs_for_repos(all_repos):
         print(f"Generating graphs for repo {repo.name}")
         generate_solid_gauge_issue_graph(repo)
         generate_repo_sparklines(repo)
+        generate_donut_graph_line_complexity_graph(repo)
```
Comment (decause-gov) [inline scripts/gen_graphs.py:19]: Absolutely love that this is how easy it is to add new graphs to generate <3

File: scripts/gen_graphs.py [hunk 2]
```diff
@@ -91,6 +92,7 @@ def generate_time_xy_issue_graph(oss_entity,data_key):
     """
 
     graph_data_dict = oss_entity.metric_data[data_key]
+    print(graph_data_dict)
```
Comment (decause-gov) [inline scripts/gen_graphs.py:95]: leftover debugging print statement?

File: scripts/gen_graphs.py [hunk 3]
```diff
@@ -101,6 +103,31 @@ def generate_time_xy_issue_graph(oss_entity,data_key):
     write_repo_chart_to_file(oss_entity, xy_time_issue_chart, data_key)
 
 
+def generate_donut_graph_line_complexity_graph(oss_entity):
+    """
+    This function generates pygals line complexitydonut graph
```
Comment (decause-gov) [inline scripts/gen_graphs.py:108]: s/complexitydonut/complexity donut/g

File: scripts/gen_graphs.py [hunk 4]
```diff
@@ -101,6 +103,31 @@ def generate_time_xy_issue_graph(oss_entity,data_key):
...
+    for a set of Repository objects.
+
+    Arguments:
+        oss_entity: The OSSEntity to create a graph for.
```
Comment (decause-gov) [inline scripts/gen_graphs.py:112]: IDEA: What is an OSSEntity? Perhaps we could say what the value typically is, where it comes from, what it is used for?

File: scripts/gen_reports.py [hunk 5]
```diff
@@ -48,6 +48,7 @@ def get_heading_report_values(headings, oss_entity):
 
     report_values = {}
     for heading in headings:
+        print(oss_entity.metric_data)
```
Comment (decause-gov) [inline scripts/gen_reports.py:51]: leftover debug print statement?

File: scripts/metricsLib/metrics_data_structures.py [hunk 6]
```diff
@@ -316,6 +316,8 @@ class ListMetric(BaseMetric):
     def __init__(self, name, needed_params, endpoint_url, return_values, token=None, method='GET'):
         super().__init__(name, needed_params, endpoint_url,
                          return_values, token=token, method=method)
+        
+        self.tuple_flag = True
```
Comment (decause-gov) [inline scripts/metricsLib/metrics_data_structures.py:320]: 🏁 🧠

File: templates/repo_report_template.md [hunk 7]
```diff
@@ -114,5 +114,10 @@ date_stampLastWeek: {date_stamp}
         <img alt="firstResponseForClosedPR" src="{{{{ "/assets/img/graphs/{repo_owner}/{repo_name}/firstResponseForClosedPR_{repo_name}_data.png" | url }}}}" />
       </figure>
     </div>
+    <!--- Line Complexity Graphs -->
```
Comment (decause-gov) [inline templates/repo_report_template.md:117]: Love that this is how easy it is to add a new graph to a report template! <3

**Review by decause-gov — COMMENTED** [review]
Reviewdog and Pylint have some solid suggestions, and I made a suggestion or two, and asked a question or threee, but, I don't see anything in here that should block a merge.

LGTM +1 🚢 🍩

**Review by decause-gov — DISMISSED** [review]

**Review by natalialuzuriaga — CHANGES_REQUESTED** [review]
Ran `./update.sh` and all LGTM! Is it possible to decrease the width of the donut so that the legend text is not cut off?

**Comment by IsaacMilarky** [comment]
> Ran `./update.sh` and all LGTM! Is it possible to decrease the width of the donut so that the legend text is not cut off?

That should be possible I will look into it

File: scripts/gen_graphs.py (hunk 4 above)
Comment (IsaacMilarky) [reply scripts/gen_graphs.py:112]: I added something about what is is typically to the docstring

File: scripts/gen_graphs.py (hunk 1 above)
Comment (IsaacMilarky) [reply scripts/gen_graphs.py:19]: I'm trying to make it as maintainable as possible

**Comment by IsaacMilarky** [comment]
@natalialuzuriaga Here is what the graph looks like now:

total_line_makeup_metrics_data

**Review by natalialuzuriaga — APPROVED** [review]
That's better!!! LGTM 🔥

**Pull request merged**